process/process.go

```diff
@@ -1,145 +1,187 @@
 package process
 
 import (
+	"os"
+	"path"
+	"strconv"
 	"time"
 
 	log "github.com/Sirupsen/logrus"
+	"github.com/asdine/storm"
+	"github.com/asdine/storm/index"
+	"github.com/asdine/storm/q"
 	"github.com/fredli74/lockfile"
 	"github.com/resin-io/edge-node-manager/application"
 	"github.com/resin-io/edge-node-manager/config"
+	"github.com/resin-io/edge-node-manager/device"
 	deviceStatus "github.com/resin-io/edge-node-manager/device/status"
 	processStatus "github.com/resin-io/edge-node-manager/process/status"
 	"github.com/resin-io/edge-node-manager/radio/bluetooth"
+	"github.com/resin-io/edge-node-manager/supervisor"
+	tarinator "github.com/verybluebot/tarinator-go"
 )
 
 var (
-	delay          time.Duration
-	CurrentStatus  processStatus.Status
-	TargetStatus   processStatus.Status
-	UpdatesPending bool
-	Lock           *lockfile.LockFile
+	pauseDelay    time.Duration
+	CurrentStatus processStatus.Status
+	TargetStatus  processStatus.Status
+	lockLocation  string
+	lock          *lockfile.LockFile
 )
 
-// Run processes the application, checking for new commits, provisioning and updating devices
-func Run(a *application.Application) []error {
-	log.Info("----------------------------------------------------------------------------------------------------")
-
-	// Put all provisioned devices associated with this application
-	defer a.PutDevices()
+func Run(a application.Application) []error {
+	log.Info("----------------------------------------")
```

Comment:
Nope, its there to highlight the change between each managed application in the logs. It's quite hard to parse without it. For example:

With the dashes:
```
23.03.17 10:48:48 (+0000) INFO ----------------------------------------
23.03.17 10:48:48 (+0000) INFO Processing application                        Application=test1
23.03.17 10:48:58 (+0000) INFO Processing application                        Number of online devices=0
23.03.17 10:48:58 (+0000) INFO Processing application                        Number of provisioned devices=2
23.03.17 10:49:12 (+0000) INFO ----------------------------------------
23.03.17 10:49:12 (+0000) INFO Processing application                        Application=test2
23.03.17 10:49:22 (+0000) INFO Processing application                        Number of online devices=0
23.03.17 10:49:22 (+0000) INFO Processing application                        Number of provisioned devices=2
```

Without the dashes:
```
23.03.17 10:48:48 (+0000) INFO Processing application                        Application=test1
23.03.17 10:48:58 (+0000) INFO Processing application                        Number of online devices=0
23.03.17 10:48:58 (+0000) INFO Processing application                        Number of provisioned devices=2
23.03.17 10:49:12 (+0000) INFO Processing application                        Application=test2
23.03.17 10:49:22 (+0000) INFO Processing application                        Number of online devices=0
23.03.17 10:49:22 (+0000) INFO Processing application                        Number of provisioned devices=2
```

I am open to other suggestions